packages/adblocker/src/filters/network.ts

```diff
@@ -400,18 +397,169 @@ function compileRegex(
+  let slashes = 0;
+
+  for (; pos < end; pos++) {
+    code = line.charCodeAt(pos);
+
+    if (code === 92 /* '\\' */) {
+      parts[slashes] += '\\' + line.charAt(++pos);
+    } else if (code === 47 /* '/' */) {
+      if (++slashes === 2) {
+        // Skip the last slash character
+        // Since we saw 3 slashes in total, it means that the option value should be closed here.
+        // Note that we already saw the first slash before the loop.
+        pos++;
+
+        break;
+      }
+    } else {
+      parts[slashes] += line.charAt(pos);
+    }
+  }
+
+  const valueEnd = line.indexOf(',', pos);
 
-  if (lastIndex === -1) {
-    return -1;
+  if (valueEnd !== -1) {
+    end = valueEnd;
   }
 
-  while (lastIndex > 0 && text.charCodeAt(lastIndex - 1) === 92 /* '\\' */) {
-    lastIndex = text.lastIndexOf(character, lastIndex - 1);
+  if (pos - end !== 0) {
+    parts[2] = line.slice(pos, end);
   }
 
-  return lastIndex;
+  pos = end;
+
+  return [pos, parts];
+}
+
+/**
+ * Collects an array of filter options from the given index.
+ * This function leverages `getFilterOptionKey`, `getFilterOptionValue`, and every extension functions.
+ * Depending on the filter option key, the function to collect filter option value can vary.
+ * For the generic filter option value, it'll use `getFilterOptionValue` function to get the value.
+ */
+function getFilterOptions(line: string, pos: number, end: number) {
+  const options: Array<[string, string]> = [];
+
+  let key: string | undefined;
+  let value: string;
+
+  for (; pos < end; pos++) {
+    [pos, key] = getFilterOptionKey(line, pos, end);
+
+    if (key !== undefined) {
+      if (line.charCodeAt(pos) === 61 /* '=' */) {
+        pos++;
+      }
+
+      if (key === 'replace') {
+        const result = getFilterReplaceOptionValue(line, pos, end);
+
+        pos = result[0];
+
+        if (result[1].length !== 0) {
+          value = '/' + result[1][0] + '/' + result[1][1] + '/' + result[1][2];
+        } else {
+          value = '';
+        }
+      } else {
+        [pos, value] = getFilterOptionValue(line, pos, end);
+      }
+
+      options.push([key, value]);
+    }
+  }
+
+  return options;
+}
+
+/**
+ * Transforms the replace modifier option value into the regular expression with its replacement.
+ * This function takes a fixed length array from `getFilterReplaceOptionValue`
+ * then try to build the regular expression.
+ * This function will return `null` if the array format or the given regular expression components are not valid.
+ */
+export function replaceOptionValueToRegexp(value: string): HTMLModifier | null {
+  const [, values] = getFilterReplaceOptionValue(value, 0, value.length);
+
+  // RegExp constructor can throw an error
+  try {
+    // We expect `/regexp/replacement/flags` to be [regexp, replacement, flags]
+    // The first slash should be removed in the early steps
+    return [new RegExp(values[0], values[2]), values[1]];
+  } catch (error) {
+    return null;
```

Comment:
I wonder if there is a way to log this as an error (without spamming)? If so, perhaps we should log the error?

Reply:
We currently don't express the parsing errors in the parsing method. I'd like to extend this idea in the separate PR instead: https://github.com/ghostery/adblocker/pull/4108.